server/controllers/oauth.py

```diff
@@ -75,7 +78,25 @@ def save_token(token, orequest, *args, **kwargs):
     db.session.commit()
     return tok
 
+def intercept_out_of_band_redirect(f):
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        response = make_response(f(*args, **kwargs))
+        if response.status_code == 302:
```

Comment:
Yes, except it should never return a 400 (always a 200 or 302). If an error occurs, it'll 302 with an error instead of a code.